### PR TITLE
work on castling rights

### PR DESCRIPTION
**Pull request by Dillon-Roller**

### CODE REVIEW

File: src/game.rs [hunk 1]
```diff
@@ -58,12 +58,18 @@ impl Game {
...
+        &self.state
+    }
+
     pub fn move_piece(&mut self, from: Position, to: Position) -> ChessResult {
         if let Some(piece) = self.state.get_piece(&from) {
             let mv = Move::new(from, to);
 
             self.is_move_valid(&mv)?;
             self.state.move_piece(&mv);
+            self.handle_capturing_en_passant(&mv.to);
```
Comment (DanielAlbl) [inline src/game.rs:71]: I think these should still be in board state

File: src/castling_rights.rs [hunk 2]
```diff
@@ -73,18 +92,52 @@ impl CastlingRights {
...
+        for (position, piece, kingside_rights, queenside_rights) in CastlingRights::king_positions()
+        {
+            if board.get_piece(&position) != Some(piece) {
+                self.0[kingside_rights as usize] = false;
+                self.0[queenside_rights as usize] = false;
+            }
+        }
+    }
+
+    pub fn handle_castling_the_rook(&mut self, mv: &Move, board: &mut Board, player: Player) {
```
Comment (DanielAlbl) [inline src/castling_rights.rs:126]: Does `self` need to be `mut`?

File: src/castling_rights.rs [hunk 3]
```diff
@@ -1,13 +1,28 @@
-use crate::pieces::{Piece, Player, Position};
+use std::collections::HashSet;
```
Comment (DanielAlbl) [inline src/castling_rights.rs:1]: Don't need this

File: src/castling_rights.rs (hunk 2 above)
Comment (Dillon-Roller) [reply src/castling_rights.rs:126]: yeah because we're setting the piece

**Review by DanielAlbl — APPROVED** [review]

**Pull request merged**